### DISCUSSION TITLE
UDUNITS usage in Argo

### DISCUSSION THREAD

**Discussion by vpaba** [OneArgo/ArgoVocabs#25] [General]
@nvs-vocabs/avtt: Argo NetCDF files use [UDUNITS](https://www.unidata.ucar.edu/software/udunits/) to populate the variable attribute 'unit' in the data section, as stated in the Argo User Manual V3.4.  

Current issues:
- There currently isn't a controlled vocabulary for UDUNITS.
- Differences between UDUNITS and [P06](http://vocab.nerc.ac.uk/collection/P06/current/) have been identified.
- Instances of inconsistent usage of UDUNITS in Argo files (particularly BGC) have been identified.

**Comment by apswong** [comment]
Hello, just a comment on the first bullet point: I have worked with Thierry to state clearly in the Argo Users Manual that Argo data use the UDUNITS convention.  You can see it in Section 2.1 (page 16 in the most recent version of the Users Manual, V3.4, dated 19 January 2021).

**Reply by vpaba** [reply]
Hi @apswong, you're right, thank you - I've updated the original post.

**Comment by catsch** [comment]
Hi @vpaba 
regarding bullet 3., do you mention that for the PAR units ?
we are thinking of moving from  µmol quanta/m2/sec to µmol/m2/sec (less informative according to us) 
We are not very keen on using Watt for PAR

**Reply by vpaba** [reply]
Hi @catsch,
I think this point was raised mostly with regards to the use of units in the Oxygen cookbook, which @matdon17 has recently reviewed.
I don't think it refers to units mismatch between R03 codes and P01 codes (such as for DOWNWELLING_PAR and DWIRRXUD) - this is a separate point which may not necessarily raise issues within Argo, though possibly with mappings to NVS collections outside of Argo.
Matt and I are planning to further discuss the topic of UDUNITS in Argo in the next couple of weeks, so we should have more insight on this soon.

**Comment by tcarval** [comment]
There are 12 parameters in table R03 having non UDUNITS units (5 units).
We should probably revisit the sentence in the "ARgo user's manual": 

- Replace : _"Argo data use the UDUNITS convention to populate the variable attribute "unit" in the Data section."_
- With : _"Argo data use the UDUNITS convention **when applicable** to populate the variable attribute "unit" in the Data section."_


parameter name | long_name | unit
-- | -- | --
PSAL | Practical   salinity | psu
NB_SAMPLE | Number   of samples in bin | dimensionless
NB_SAMPLE_CTD | Number   of samples in each pressure bin | dimensionless
NB_SAMPLE_SFET | Number   of samples in each pressure bin for the SFET | dimensionless
NB_SAMPLE_&lt;short_sensor_name&gt; | Number   of samples in each pressure bin for the &lt;short_sensor_name&gt; | dimensionless
TURBIDITY | Sea   water turbidity | ntu
CHLA | Chlorophyll-A | mg/m3
FIT_ERROR_NITRATE | Nitrate   fit error | dimensionless
PH_IN_SITU_TOTAL | pH | dimensionless
PH_IN_SITU_FREE | pH | dimensionless
PH_IN_SITU_SEAWATER | pH | dimensionless
DOWNWELLING_PAR | Downwelling   photosynthetic available radiation | microMoleQuanta/m^2/sec

**Reply by apswong** [reply]
Hi @tcarval, I think editing that sentence in the Argo Users Manual is a good idea.

**Comment by delphinedobler** [comment]
The sentence was modified within UM3.41 version
